### ISSUE TITLE
`stylistic.configs.customize({ quotes: "backtick" })` gives type error even though it's valid

### ISSUE THREAD

**Issue by samualtnorman** [eslint-stylistic/eslint-stylistic#568]
### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

`eslint.config.js`
```js
import stylistic from "@stylistic/eslint-plugin"

export default [ stylistic.configs.customize({ quotes: "backtick" }) ]
```
reports
```
No overload matches this call.
  Overload 1 of 2, '(options: StylisticCustomizeOptions<false>): BaseConfig<RulesRecord, RulesRecord>', gave the following error.
    Type '"backtick"' is not assignable to type '"single" | "double" | undefined'.
  Overload 2 of 2, '(options?: StylisticCustomizeOptions<true> | undefined): Config<RulesRecord>', gave the following error.
    Type '"backtick"' is not assignable to type '"single" | "double" | undefined'.ts(2769)
```
even though it's a valid option

### Reproduction

to reproduce, use the config from above

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests
